hs_file_types/models/base.py

```diff
@@ -660,6 +927,65 @@ def supports_unzip(self):
         """allows a zip file that is part of this logical file type to get unzipped"""
         return True
 
+    @property
+    def aggregation_name(self):
+        """Returns aggregation name as per the aggregation naming rule defined in issue#2568"""
+        if self.is_single_file_aggregation:
+            # self is a single file aggregation type
+            return self.files.first().short_path
+        else:
+            # self is a multi- file aggregation type
+            return self.files.first().file_folder
+
+    @property
+    def metadata_short_file_path(self):
+        """File path of the aggregation metadata xml file relative to {resource_id}/data/contents/
+        """
+
+        xml_file_name = self.aggregation_name
+        if "/" in xml_file_name:
+            _, xml_file_name = os.path.split(xml_file_name)
+        xml_file_name += "_meta.xml"
+        file_folder = self.files.first().file_folder
+        if file_folder is not None:
+            xml_file_name = os.path.join(file_folder, xml_file_name)
+        return xml_file_name
+
+    @property
+    def metadata_file_path(self):
+        """Full file path of the aggregation metadata xml file starting with {resource_id}/data/contents/
+        """
+        return os.path.join(self.resource.file_path, self.metadata_short_file_path)
+
+    @property
+    def map_short_file_path(self):
+        """File path of the aggregation map xml file relative to {resource_id}/data/contents/
+        """
+        xml_file_name = self.aggregation_name
+        if "/" in xml_file_name:
+            _, xml_file_name = os.path.split(xml_file_name)
+        xml_file_name += "_resmap.xml"
+        file_folder = self.files.first().file_folder
+        if file_folder is not None:
+            xml_file_name = os.path.join(file_folder, xml_file_name)
+        return xml_file_name
+
+    @property
+    def map_file_path(self):
+        """Full file path of the aggregation map xml file starting with {resource_id}/data/contents/
+        """
+        return os.path.join(self.resource.file_path, self.map_short_file_path)
+
+    @property
+    def is_single_file_aggregation(self):
+        """
+        Returns True if the aggregation consists of only one file, otherwise, False.
+        Subclasses that support only single file must override this property
+
+        :return: True or False
+        """
+        return False
```

Comment:
raise NotImplemented?

Reply:
With a return of False here, only the multi-file aggregation types need to override this as the doc says.